lib/supavisor/monitoring/telem.ex

```diff
@@ -48,12 +48,17 @@ defmodule Supavisor.Monitoring.Telem do
     )
   end
 
-  @spec client_join(:ok | :fail, S.id()) :: :ok
+  @spec client_join(:ok | :fail, S.id() | any()) :: :ok
   def client_join(status, {tenant, user, mode}) do
     :telemetry.execute(
       [:supavisor, :client, :joins, status],
       %{},
       %{tenant: tenant, user: user, mode: mode}
     )
   end
+
+  def client_join(status, _id) when not is_tuple(_id) do
+    Logger.warn("client_join called with non-tuple id: #{inspect(_id)}")
+    :ok
+  end
```

Comment:
We could handle all cases "other" than the one mentioned in the clause above, ensuring that this function will never fail. Additionally, since [`warn`](https://github.com/elixir-lang/elixir/blob/74bfab8ee271e53d24cb0012b5db1e2a931e0470/lib/logger/lib/logger.ex#L866) is deprecated, we need to use the full term `warning`. It returns [`:ok`](https://hexdocs.pm/logger/1.12.3/Logger.html#warning/2)
```suggestion
  def client_join(status, id) do
    Logger.warning("client_join is called with a mismatched id: #{inspect(id)}")
  end
```